### ISSUE TITLE
Unshuffled UTXO are moved to new wallet even when spened shuffled is activated

### ISSUE THREAD

**Issue by damascene** [KarolTrzeszczkowski/Inter-Wallet-Transfer-EC-plugin#5]
Today I was testing the plugin with cashshuffle enabled. I've noticed that an unshuffled UTXO were send to the new wallet. I'm not sure if this could be a privacy issue but I think that the Inter Wallet Transfer plugin should honour that switch and shouldn't send those UTXOs

![resim](https://user-images.githubusercontent.com/411046/61129808-e2103d00-a4bd-11e9-852c-8330c5d0765e.png)

**Comment by KarolTrzeszczkowski** [comment]
Yes, it certainly should. Thank you for this catch. I'll fix it. Note, that transferring between wallets when shuffle is turned on will cause the plugin to crash. 

Fortunately, it won't spoil your privacy, the address is fresh and never used. You can shuffle the coin now.

**Comment by damascene** [comment]
the plugin crashed several times indeed, it was giving me tones of error with this inside it:
Transaction could not be broadcast due to dust outputs (dust threshold is 546 satoshis).

**Comment by cculianu** [comment]
@damascene The crash bugs should now be fixed in latest.. we added dust checking and other checks. :)

**Comment by molecular** [comment]
is it possible to make the plugin only transfer coins that went through CashFusion?

**Comment by KarolTrzeszczkowski** [comment]
I'll have to ask around, I don't know fusion internal implementation, how they mark already fused coins.

**Comment by cculianu** [comment]
They aren’t marked as such, unfortunately.

As the plugin maintainer — You would have to do some digging into the tx that generated the fusion looking for the FUZ op_return and the Fusion label (maybe) to have reasonable confidence it was a fused coin. And you would have to do that *for every coin*... so it might get slow...


Even then it’s not Guaranteed to be perfect...

**Comment by molecular** [comment]
maybe that sub-account idea isn't that bad after all to get some order into utxos
